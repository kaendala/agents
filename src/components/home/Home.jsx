import "./Home.scss";
import React, { useState } from "react";
import { useDispatch } from "react-redux";
import { changeIncome } from "../../redux/actions/income.action";
import { formatNumber, removeCurrencyFormat } from "../../utils/number";
import { useHistory } from "react-router-dom";

const Home = () => {
  const history = useHistory();
  const dispach = useDispatch();
  const [error, setError] = useState(null);
  const [inputValue, setInputValue] = useState("");

  const inputChange = (evt) => {
    const regex = /[0-9]*/;
    evt.target.value = removeCurrencyFormat(evt.target.value);
    const validValues = regex.exec(evt.target.value);
    setInputValue(formatNumber("en-US", "USD", 0, validValues[0]));
    if (evt.target.value.length !== 5) {
      setError(true);
    } else {
      setError(false);
    }
  };
  return (
    <div className="home">
      <div className="contentImage">
        <img alt="person" src="/assets/images/Vector.png"></img>
      </div>
      <h1 className="title">Find the best agent for you!</h1>
      <h3 className="subtitle">
        Fill the information below to get your matches.
      </h3>
      <div id="incomeForm" className="contentInput">
        <h4 className="label">Current income</h4>
        <div className="iconinput">
          <span className="fas fa-dollar-sign"></span>
          <input
            type="text"
            required={true}
            id="income"
            maxLength={6}
            minLength={6}
            onChange={inputChange.bind(this)}
            autoComplete="off"
            value={inputValue}
          ></input>
        </div>

        <h4 className={"error" + (error ? " show" : "")}>
          {"Must be 5 numbers"}
        </h4>
        <div className="contentButton">
          <button
            disabled={error || error === null}
            onClick={() => {
              dispach(changeIncome(parseInt(removeCurrencyFormat(inputValue))));
              history.push("/agents");
            }}
          >
            Get matches<span className="fas fa-arrow-right"></span>
          </button>
        </div>
      </div>
    </div>
  );
};

export default Home;
