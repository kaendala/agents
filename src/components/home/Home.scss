@import "../../../src/style/global.scss";
.home {
  width: 100%;
  .contentImage {
    width: 100%;
    img {
      width: 6.6%;
      max-width: 91px;
      margin-top: 10.7%;
    }
  }
  .title {
    margin-top: 24.5px;
  }
  .subtitle {
    margin-top: 16px;
    font-weight: 400;
    font-size: 16px;
    color: $GD600;
  }

  .contentInput {
    margin: 0 auto;
    width: 23.43%;
    max-width: 322px;
    .label {
      margin: 56px 0 9px;
      width: 100%;
      text-align: left;
    }
    .iconinput {
      position: relative;
      display: flex;
      align-items: center;
      .fas {
        position: absolute;
        left: 4.4%;
      }
    }
    input {
      width: 100%;
      height: 46px;
      padding: 1px 7% 0;
      border: 1px solid $GL300;
      box-sizing: border-box;
      border-radius: 4px;
      outline: none;
      font-size: 18px;
    }
    .error {
      visibility: hidden;
      font-size: 12px;
      font-style: italic;
      text-align: left;
      color: $R600;
      &.show {
        visibility: visible;
      }
    }
    .contentButton {
      display: flex;
      justify-content: flex-end;
      margin-top: 28px;
      button {
        appearance: none;
        height: 46px;
        width: 55%;
        background-color: $B03;
        color: $W0;
        border-radius: 4px;
        border: none;
        font-size: 16px;
        font-weight: 700;
        display: flex;
        align-items: center;
        justify-content: space-around;
        &:disabled {
          background-color: $GD400;
        }
      }
    }
  }
}
@media (max-width: 768px) {
  .home {
    .contentImage {
      img {
        width: 24%;
        margin-top: 20.4%;
      }
    }
    .title {
      width: 82%;
      margin: 36px auto 0;
    }
    .subtitle {
      width: 82%;
      margin: 16px auto 0;
    }

    .contentInput {
      width: 82%;
      .label {
        margin-top: 32px;
      }
      .contentButton {
        button {
          width: 100%;
          position: relative;
          span {
            position: absolute;
            right: 5%;
          }
        }
      }
    }
  }
}
