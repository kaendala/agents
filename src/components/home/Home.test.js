import React from 'react';
import Home from './Home';
import configureMockStore from 'redux-mock-store';
import { fireEvent,render } from '@testing-library/react';
import { Provider } from 'react-redux';

const mockStore = configureMockStore();
const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('Test for Home component', () => {
    let component;
    const storeInitialState = mockStore({
      incomeReducer: {
        income: 20000
        }
    });
    beforeEach(async() => {
            component = render(<Provider store={storeInitialState}><Home></Home></Provider>);
        })
    test('should load component Home ', () => {
        expect(component).toMatchSnapshot();
    });
    test('should call history ', () => {
        const input = component.container.querySelector("input")
        fireEvent.input(input, {
            target: {
                value: '20',
            },
        });
        const error = component.container.querySelector(".error.show")
        expect(error).toBeTruthy();
        fireEvent.input(input, {
            target: {
                value: '20000',
            },
        });
        const button = component.container.querySelector("button").click()
        expect(mockHistoryPush).toBeCalled();
    });
})