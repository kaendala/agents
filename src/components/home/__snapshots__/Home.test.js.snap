// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Home component should load component Home  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <div
          class="contentImage"
        >
          <img
            alt="person"
            src="/assets/images/Vector.png"
          />
        </div>
        <h1
          class="title"
        >
          Find the best agent for you!
        </h1>
        <h3
          class="subtitle"
        >
          Fill the information below to get your matches.
        </h3>
        <div
          class="contentInput"
          id="incomeForm"
        >
          <h4
            class="label"
          >
            Current income
          </h4>
          <div
            class="iconinput"
          >
            <span
              class="fas fa-dollar-sign"
            />
            <input
              autocomplete="off"
              id="income"
              maxlength="6"
              minlength="6"
              required=""
              type="text"
              value=""
            />
          </div>
          <h4
            class="error"
          >
            Must be 5 numbers
          </h4>
          <div
            class="contentButton"
          >
            <button
              disabled=""
            >
              Get matches
              <span
                class="fas fa-arrow-right"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <div
        class="contentImage"
      >
        <img
          alt="person"
          src="/assets/images/Vector.png"
        />
      </div>
      <h1
        class="title"
      >
        Find the best agent for you!
      </h1>
      <h3
        class="subtitle"
      >
        Fill the information below to get your matches.
      </h3>
      <div
        class="contentInput"
        id="incomeForm"
      >
        <h4
          class="label"
        >
          Current income
        </h4>
        <div
          class="iconinput"
        >
          <span
            class="fas fa-dollar-sign"
          />
          <input
            autocomplete="off"
            id="income"
            maxlength="6"
            minlength="6"
            required=""
            type="text"
            value=""
          />
        </div>
        <h4
          class="error"
        >
          Must be 5 numbers
        </h4>
        <div
          class="contentButton"
        >
          <button
            disabled=""
          >
            Get matches
            <span
              class="fas fa-arrow-right"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
